server/pages/serviceWorker.js

```diff
@@ -0,0 +1,141 @@
+// Taken from:
+// https://github.com/GoogleChrome/samples/blob/e4df12c8642381243b6c1710c41394d85b33d82f/service-worker/prefetch/service-worker.js
+
+// While overkill for this specific sample in which there is only one cache,
+// this is one best practice that can be followed in general to keep track of
+// multiple caches used by a given service worker, and keep them all versioned.
+// It maps a shorthand identifier for a cache to a specific, versioned cache name.
+
+// Note that since global state is discarded in between service worker restarts, these
+// variables will be reinitialized each time the service worker handles an event, and you
+// should not attempt to change their values inside an event handler. (Treat them as constants.)
+
+// If at any point you want to force pages that use this service worker to start using a fresh
+// cache, then increment the CACHE_VERSION value. It will kick off the service worker update
+// flow and the old cache(s) will be purged as part of the activate event handler when the
+// updated service worker is activated.
+var CACHE_VERSION = 1;
+var CURRENT_CACHES = {
+	prefetch: 'calypso-prefetch-cache-v' + CACHE_VERSION
+};
+
+self.addEventListener('install', function( event ) {
```

Comment:
Now you could `self.skipWaiting()` to not wait until other clients being controlled by other sw close. This, in combination with `self.clients.claim()` during activation leads to controlling the clients ASAP.

See more about the life-cycle here: https://github.com/delapuente/service-workers-101/